File: Python/2-Medium/00045._跳跃游戏_II/00045._跳跃游戏_II.py
from typing import List
def jump(nums: List[int]) -> int:
    """
    使用贪心算法求解跳跃游戏 II 的最小跳跃次数。

    Args:
    nums: 一个非负整数数组，表示可以从当前位置跳跃的最大长度。

    Returns:
    int: 到达数组最后一个位置的最小跳跃次数。

    示例:
    >>> jump([2,3,1,1,4])
    2
    """
    jumps, furthest, end = 0, 0, 0
    for i in range(len(nums) - 1):  # 遍历数组，但不包括最后一个元素，因为跳跃动作在到达最后一个元素前完成
        furthest = max(furthest, i + nums[i])  # 更新能够到达的最远距离
        if i == end:  # 如果当前位置达到了这一跳的最远距离
            jumps += 1  # 进行一次跳跃
            end = furthest  # 更新下一次跳跃的最远距离
    return jumps

# 测试代码
test_cases = [
    [2, 3, 1, 1, 4],  # 示例 1
    [2, 3, 0, 1, 4],  # 示例 2
]

results = [jump(case) for case in test_cases]
print(results)  # [2, 2]

File: C++/2-Medium/00045._跳跃游戏_II/00045._跳跃游戏_II.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int jump(vector<int>& nums) {
        int jump = 0, maxReach = 0, end = 0;
        // 遍历数组中的每个元素，除了最后一个元素，因为一旦到达最后一个元素，就不需要再跳跃了
        for (int i = 0; i < nums.size() - 1; i++) {
            // 更新能到达的最远距离
            maxReach = max(maxReach, i + nums[i]);
            // 如果到达当前探索的边界
            if (i == end) {
                // 进行跳跃，更新探索的边界
                end = maxReach;
                jump++;
            }
        }
        return jump;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {2, 3, 1, 1, 4};
    cout << "Minimum jumps required: " << solution.jump(nums) << endl;
    return 0;
}